src/_posts/2020-03-24-project-maturity-model.md

```diff
@@ -0,0 +1,153 @@
+Errors were still reported by our users.
+
+We started to fix the error levels of the messages, Some errors were merely informational and could be downgraded. 
+Others needed more details.
+
+We added [Dependabot](/2019/02/24/taming-dependabot/) to all of our projects to assist with upgrades. This allows us to keep on top of the security and other changes to the libararies that we depend upon.
+
+### Second Crisis
+
+We started to get complaints that our users were not recieving emails.
+This was odd as we could see in the logs that some emails were being sent out.
+
+We used datadog to produce a dashboard that produces graphs of both the signals that an email should be sent along with a list of the number of emails actually sent.
+This was enough for us to see that while we were sending some emails, it was significantly lower than expected.
+
+One of our internal services had changed it's contract and one of the services could not cope with the new shape of the data. This did not take long to solve or isolate given the now improving diagrams.
+
+### Level 3 Defined
+
+We now started to reduce the most frequent group of errors whenever we had a quiet time. Datadog has a great option to detect patterns in the logs and create groups out of them. 
+These form obvious targets for reducing the errors. We started adding alerts for specific problems to ensure that we know about a problem before the users tell us.
+
+We were still occasionally getting problems reported just before hometime on a Friday. Solving these is much quicker now.
+
+
+## Third Crisis
+Our QA while testing a change noticed a message on a database dashboard. The database service that we were using for eight of the services is being decomissioned in 6 months time.
+We fairly quickly find that four of the databases are actually only used for locking and were able to replace them with AWS Lamdas. The remaining four we eventually migrated.
+
+The diagrams are getting more accurate by this point.
+
+## Further Improvements
+
+We added integration tests to check that the access codes that we needed for each remote service were still accurate. We are integrating with over 100 almost identical services and they are undergoing an upgrade program! This means that the tests will be broken first thing on Monday morning, allowing us to fix things before our users need the data.
+
+One of the services used AWS S3 to store messages that could not be parsed. Expanding this to other parts of the service greatly assisted fixing (or at least quieting) some strange errors.
+
+One of the services was responsible for querying an external service (which had over 100 instances). At any given time some of these were broken (2% to 5%, but not always the same one). For a while we were moving the broken feeds to use a less effictive backup service manually. This required us to remember to restore the feeds once they had recovered.
```

Comment:
Change sentence: For a while, we were moving the broken feeds to use a less effective backup service manually.